src/shipchain_common/authentication.py

```diff
@@ -43,19 +43,20 @@ def get_jwt_from_request(request):
     return None
 
 
-def is_internal_call(request):
-    return ('X_NGINX_SOURCE' in request.META and request.META['X_NGINX_SOURCE'] == 'internal'
-            and request.META['X_SSL_CLIENT_VERIFY'] == 'SUCCESS')
+def is_internal_call(request, service_name=None):
+    is_internal = ('X_NGINX_SOURCE' in request.META and request.META['X_NGINX_SOURCE'] == 'internal'
+                   and request.META['X_SSL_CLIENT_VERIFY'] == 'SUCCESS')
+    if service_name and is_internal:
+        certificate_cn = parse_dn(request.META['X_SSL_CLIENT_DN'])['CN']
+        is_internal = is_internal and certificate_cn == f'{service_name}.{settings.ENVIRONMENT.lower()}-internal'
```

Comment:
At this point `is_internal` is guaranteed to be `True`.  No need to include `is_internal and` at the beginning of this assignment.

Reply:
I originally had it set up this way to be very explicit in needing the `is_internal` to be considered true